packages/manager/apps/pci-kubernetes/src/components/Autoscaling.component.tsx

```diff
@@ -0,0 +1,183 @@
+import {
+  ODS_TEXT_COLOR_INTENT,
+  ODS_TEXT_LEVEL,
+  ODS_ICON_NAME,
+  ODS_ICON_SIZE,
+  ODS_MESSAGE_TYPE,
+} from '@ovhcloud/ods-components';
+import { OdsHTMLAnchorElementTarget } from '@ovhcloud/ods-common-core';
+import {
+  OsdsFormField,
+  OsdsIcon,
+  OsdsLink,
+  OsdsMessage,
+  OsdsText,
+  OsdsToggle,
+} from '@ovhcloud/ods-components/react';
+import {
+  ODS_THEME_COLOR_INTENT,
+  ODS_THEME_TYPOGRAPHY_SIZE,
+} from '@ovhcloud/ods-common-theming';
+import { useTranslation } from 'react-i18next';
+import { useMe } from '@ovhcloud/manager-components';
+import { useEffect, useState } from 'react';
+import { QuantitySelector } from '@ovh-ux/manager-pci-common';
+
+const AUTOSCALING_LINK = {
+  DEFAULT: 'https://docs.ovh.com/gb/en/kubernetes/using-cluster-autoscaler/',
+  US: 'https://support.us.ovhcloud.com/hc/en-us/articles/1500009150301',
+};
+
+const NODE_RANGE = {
+  MIN: 3,
+  MAX: 100,
+};
+
+const ANTI_AFFINITY_MAX_NODES = 5;
+
+export interface AutoscalingState {
+  desiredQuantity: number;
+  minQuantity: number;
+  maxQuantity: number;
+  isAutoscale: boolean;
+}
+
+export interface AutoscalingProps {
+  initialScaling?: { min: number; max: number; desired: number };
+  isMonthlyBilling?: boolean;
+  isAntiAffinity?: boolean;
+  onChange?: (scaling: AutoscalingState) => void;
+}
+
+export function Autoscaling({
+  initialScaling,
+  isMonthlyBilling,
+  isAntiAffinity,
+  onChange,
+}: Readonly<AutoscalingProps>) {
+  const { t } = useTranslation('autoscaling');
+  const ovhSubsidiary = useMe()?.me?.ovhSubsidiary;
+  const infosURL = AUTOSCALING_LINK[ovhSubsidiary] || AUTOSCALING_LINK.DEFAULT;
+  const [isAutoscale, setIsAutoscale] = useState(false);
+  const [desiredQuantity, setDesiredQuantity] = useState(
+    initialScaling ? initialScaling.desired : NODE_RANGE.MIN,
+  );
+  const [minQuantity, setMinQuantity] = useState(
+    initialScaling ? initialScaling.min : 0,
+  );
+  const [maxQuantity, setMaxQuantity] = useState(
+    initialScaling ? initialScaling.max : NODE_RANGE.MAX,
+  );
```

Comment:
**niptick** : values are quite linked so you might use one state for all those values

Reply:
done